web/concrete/src/Service/Configuration/GeneratorInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Concrete\Core\Service\Configuration;
+
+interface GeneratorInterface
+{
+    /**
+     * Add a new rule to be handled.
+     *
+     * @param string $handle The rule handler
+     * @param string $rule The rule
+     * @param bool|callable|null $enabled Should the rule enabled (true) or disabled (false)?
+     */
+    public function addRule($handle, $rule, $enabled);
+
+    /**
+     * Returns all the defined rules. The keys are the rules handles, the values are the rule definitions.
+     * Each definition is an array with the following keys:
+     * - string 'commentBefore' [optional] An optional part that *may* be present in the configuration before the rule to be checked.
+     * - string 'code' The code of the rule
+     * - string 'commentAfter' [optional] An optional part that *may* be present in the configuration after the rule to be checked.
+     *
+     * @return array[]
```

Comment:
It'd be better if this rule was an object, that way you don't need to query the list of rules an extra time to see if a rule is enabled:

```php
$rule = $generator->getRule('pretty_urls');
if ($rule->enabled()) {
    ...
}
```